### PR TITLE
feat: Suppress envoy headers

### PR DESCRIPTION
**Pull request by liorokman**
**What this PR does / why we need it**:

This PR implements #2358 

**Which issue(s) this PR fixes**:
Implements #2358

### CODE REVIEW

**Review by shawnh2 — APPROVED** [review]
lgtm

File: internal/gatewayapi/clienttrafficpolicy.go [hunk 1]
```diff
@@ -349,3 +353,14 @@ func translateListenerProxyProtocol(enableProxyProtocol *bool, httpIR *ir.HTTPLi
 		httpIR.EnableProxyProtocol = true
 	}
 }
+
+func translateListenerSuppressEnvoyHeaders(suppressEnvoyHeaders *bool, httpIR *ir.HTTPListener) {
+	// Return early if not set
+	if suppressEnvoyHeaders == nil {
```
Comment (Xunzhuo) [inline internal/gatewayapi/clienttrafficpolicy.go:359]:
```go
**func translateListenerSuppressEnvoyHeaders(suppressEnvoyHeaders *bool, httpIR *ir.HTTPListener) {
	if suppressEnvoyHeaders != nil {
		httpIR.SuppressEnvoyHeaders = *suppressEnvoyHeaders
	}
}
```

How about simplifying it like this way?

Comment (liorokman) [reply internal/gatewayapi/clienttrafficpolicy.go]: Done.

**Review by Xunzhuo — APPROVED** [review]

**Review by zirain — APPROVED** [review]

**Comment by arkodg** [comment]
we may need to revisit where this field lives in the future and consider moving it into `BackendTrafficPolicy` if
* this feature is something that is tied to the developer persona instead of the system admin persona
* if there are valid use cases where a user would like to enable this feature at the `Gateway` level (all HTTPRoutes ) by default but disable it for some explicit routes (using a explicit `BackendTrafficPolicy` targeted at the route level)

**Pull request merged**